### PR TITLE
Add use case tests

### PR DESCRIPTION
**Pull request by ealush**
Adding full use case tests both on the source code and on the distribution code.
More tests that cover a wider range of the api can be added. These few cover the most common uses.

### CODE REVIEW

File: src/spec/usecase/passable.usecase.spec.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { expect } from 'chai';
+import dist from '../../../dist/Passable.min';
+import dev from '../../Passable';
+import usecase_a from './usecase_a';
+import usecase_b from './usecase_b';
+import usecase_c from './usecase_c';
+
+describe('Test Passable full usecase', () => {
+
+    it('should return correct response object for case: a', () => {
```
Comment (sahariko) [inline src/spec/usecase/passable.usecase.spec.js]: Could be done in a loop

File: src/spec/usecase/usecase_a.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                inside: ['a', 'b']
+            });
+        });
+
+        pass('field_5', 'Must either be a number or a string. Always smaller than 5', () => {
+            enforce('log').anyOf({
+                isNumber: true,
+                isString: true
+            }).allOf({
+                smallerThan: 5
```
Comment (sahariko) [inline src/spec/usecase/usecase_a.js]: How is this passing? Is Passable measuring String length if this test is tested against a String and not a Number?
Same for the other cases.

File: src/spec/usecase/usecase_a.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            });
+        });
+
+        pass('field_5', 'Must not be larger than 3', () => {
```
Comment (sahariko) [inline src/spec/usecase/usecase_a.js]: Are you intentionally using `field_5` twice?
Same for the other cases.

File: src/spec/usecase/usecase_a.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        warnCount: 1,
+        testCount: 6,
+        testsPerformed: {
+            field_1: {
+                testCount: 1,
+                failCount: 0,
+                warnCount: 0
+            },
+            field_2: {
+                testCount: 1,
```
Comment (sahariko) [inline src/spec/usecase/usecase_a.js]: I think you're missing one test in the test count.

File: src/spec/usecase/usecase_a.js (hunk 2 above)
Comment (ealush) [reply src/spec/usecase/usecase_a.js]: Yes.
> Returns true if a given value is smaller than another value. The values do not have to be of the same type
https://fiverr.github.io/passable/enforce/rules/size/smaller_than/index.html

More specifically, it internally uses the `getSize` helper function which determines the size/length of an item.

https://github.com/fiverr/passable/blob/master/src/enforce/rules/helpers/get_size/index.js

File: src/spec/usecase/usecase_a.js (hunk 3 above)
Comment (ealush) [reply src/spec/usecase/usecase_a.js]: Yes, Passable allows writing multiple tests for the same field. Some of which may fail and some may pass. Here I am testing its ability to do so correctly.

File: src/spec/usecase/usecase_a.js (hunk 4 above)
Comment (ealush) [reply src/spec/usecase/usecase_a.js]: test per `pass`, not per each internal test being performed.
You could think of it as:
one test per each possible error return.

Comment (sahariko) [reply src/spec/usecase/usecase_a.js]: If so, then the usage in other cases is wrong.
For example - `field_3` in use case C.

**Pull request merged**